crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -462,13 +462,6 @@ impl Analyzer<'_, '_> {
                                     )));
                                 }
 
-                                if expanded_ty.is_query() {
-                                    unreachable!(
```

Comment:
Why did you remove this? I'm a bit worried about a stack overflow

Reply:
```ts
var g: { x: typeof g };
var g: typeof g.x;
```
`g.x` is `typeof g`

Eventually, the infinite recursion will get any from normalize and terminate.